pkg/util/overridemanager/overridemanager.go

```diff
@@ -190,7 +190,20 @@ func (o *overrideManagerImpl) getOverridersFromOverridePolicies(policies []Gener
 			resourceMatchingPolicies = append(resourceMatchingPolicies, policy)
 		}
 	}
-
+	sort.SliceStable(resourceMatchingPolicies, func(i, j int) bool {
+		implicitPriorityI := util.ResourceMatchSelectorsPriority(resource, resourceMatchingPolicies[i].GetOverrideSpec().ResourceSelectors...)
+		if len(resourceMatchingPolicies[i].GetOverrideSpec().ResourceSelectors) == 0 {
+			implicitPriorityI = util.PriorityMatchAll
+		}
+		implicitPriorityJ := util.ResourceMatchSelectorsPriority(resource, resourceMatchingPolicies[j].GetOverrideSpec().ResourceSelectors...)
+		if len(resourceMatchingPolicies[j].GetOverrideSpec().ResourceSelectors) == 0 {
+			implicitPriorityJ = util.PriorityMatchAll
+		}
+		if implicitPriorityI != implicitPriorityJ {
```

Comment:
```go
implicitPriorityI := util.ResourceMatchSelectorsPriority(resource, resourceMatchingPolicies[i].GetOverrideSpec().ResourceSelectors...)

implicitPriorityJ := util.ResourceMatchSelectorsPriority(resource, resourceMatchingPolicies[j].GetOverrideSpec().ResourceSelectors...)

if implicitPriorityI != implicitPriorityJ {
...
```
I think `if len(resourceMatchingPolicies[j].GetOverrideSpec().ResourceSelectors) == 0` is not necessary